cds_dojson/marc21/fields/books/book.py

```diff
@@ -31,8 +35,62 @@
 @filter_values
 def alternative_titles(self, key, value):
     """Alternative titles."""
+    if 'n' in value:
+        self['volume'] = volume(self, key, value)
+    if 'p' in value:
+        # if series detected
+        if self.get('volumes', None):
+            val_p = clean_val('p', value, str)
+            self['volumes_titles'].append(
+                {'title': val_p, 'volume': volume(self, key, value)}
+            ) if val_p else None
+        else:
+            self['volumes_titles'] = []
+    else:
+        return {
+            'title': clean_val('a', value, str, req=True),
+            'subtitle': clean_val('b', value, str),
+            'source': clean_val('i', value, str),
+        }
+
+
+@model.over('volume', '^246__n')
+@for_each_value
+@filter_values
+def volume(self, key, value):
+    """Translates volumes index in series."""
+    _volume = self.get('volume', None)
+    val_n = clean_val('n', value, str, req=True)
+    if volume and volume != val_n:
```

Comment:
Should be like this, right?
```suggestion
    if _volume and _volume != val_n:
```